outlines/text/generate/sequence.py

```diff
@@ -156,7 +156,7 @@ def __call__(
         self,
         prompt: Union[str, List[str]],
         samples: int = 1,
-        rng: Generator = np.random.default_rng(),
+        rng: torch.Generator = torch.Generator(),
```

Comment:
This generator should also be intialised on the device. `rng: torch.Generator = torch.Generator(device=device)` should fix it.